sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-text.service.ts

```diff
@@ -115,6 +133,100 @@ export class ApplicationWithdrawalImportTextService {
+    const applicationDataMap: ApplicationDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationValidationData = await this.getApplicationValidationData(
+      applicationNumbers,
+      institutionId,
+    );
+    applicationValidationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    return textValidations.map((textValidation) => {
+      const validationModel = {} as ApplicationBulkWithdrawalValidationModel;
+      const applicationData =
+        applicationDataMap[textValidation.textModel.applicationNumber];
+      validationModel.sin = textValidation.textModel.sin;
+      validationModel.applicationNumber =
+        textValidation.textModel.applicationNumber;
+      validationModel.withdrawalDate = textValidation.textModel.withdrawalDate;
+      validationModel.applicationFound = false;
+      if (applicationData) {
+        validationModel.applicationFound = true;
+        validationModel.studentSINMatch =
+          textValidation.textModel.sin === applicationData.sin;
+        validationModel.hasCorrectInstitutionCode =
+          originator === applicationData.locationCode;
+        validationModel.isArchived = applicationData.isArchived;
+        validationModel.applicationStatus = applicationData.applicationStatus;
+        validationModel.hasPreviouslyBeenWithdrawn =
+          applicationData.hasPreviouslyBeenWithdrawn;
+      }
+      return validationModel;
+    });
+  }
+
+  /**
+   * Get the application validation details for all the applications.
+   * @param applicationNumbers application numbers for which the application details need to be retrieved.
+   * @param institutionId institution id.
+   * @returns applications containing the required information.
+   */
+  private async getApplicationValidationData(
+    applicationNumbers: string[],
+    institutionId: number,
+  ): Promise<Application[]> {
+    return this.applicationRepo.find({
+      select: {
+        id: true,
+        applicationNumber: true,
+        isArchived: true,
+        applicationStatus: true,
+        location: { id: true, institutionCode: true },
+        student: { id: true, sinValidation: { sin: true } },
+        studentScholasticStandings: {
+          id: true,
+        },
+      },
+      relations: {
+        location: true,
+        student: { sinValidation: true },
+        studentScholasticStandings: true,
+      },
+      where: {
+        applicationNumber: In(applicationNumbers),
+        location: {
+          institution: { id: institutionId },
```

Comment:
I trying to remember the decision about changing the query to use the location code from the file header in addition to the institution id. Is it still pending?
I can see that there is a message "The institution location code provided is incorrect." for the incorrect location code but should we not let that fall into the same "Not found" scenario?